### PR TITLE
ARROW-6509: [C++][Gandiva] Re-enable Gandiva JNI tests and fix Travis CI failure

### PR DESCRIPTION
**Pull request by pprudhvi**

### CODE REVIEW

**Comment by pprudhvi** [comment]
@pitrou

**Comment by fsaintjacques** [comment]
Can you describe what was the issue?

**Comment by wesm** [comment]
Seems like the mimalloc allocator interacts badly with the JVM in some way.

File: ci/travis_before_script_cpp.sh [hunk 1]
```diff
@@ -62,7 +62,10 @@ if [ "$ARROW_TRAVIS_USE_TOOLCHAIN" == "1" ]; then
   CMAKE_COMMON_FLAGS="$CMAKE_COMMON_FLAGS -DARROW_WITH_BZ2=ON"
 fi
 
-CMAKE_COMMON_FLAGS="$CMAKE_COMMON_FLAGS -DARROW_MIMALLOC=ON"
+
+if [ "$DISTRO_CODENAME" != "xenial" ]; then
+  CMAKE_COMMON_FLAGS="$CMAKE_COMMON_FLAGS -DARROW_MIMALLOC=ON"
+fi
```
Comment (wesm) [inline ci/travis_before_script_cpp.sh]: This is definitely a hack. Can you instead add an option to toggle on/off mimalloc from .travis.yml?

**Comment by pravindra** [comment]
@pprudhvi can you add the stack from the coredump (for the mimalloc case) in the jira ?

File: ci/travis_before_script_cpp.sh (hunk 1 above)
Comment (pprudhvi) [reply ci/travis_before_script_cpp.sh]: updated. Please take a look

**Review by wesm — APPROVED** [review]
+1. Thanks for adding the backtrace in the JIRA

**Pull request closed without merging**